src/Workspaces/Core/Portable/Classification/CommonClassificationService.cs

```diff
@@ -0,0 +1,120 @@
+
+        public override async Task<ImmutableArray<ClassifiedSpan>> GetSyntacticClassificationsAsync(Document document, TextSpan span, CancellationToken cancellationToken)
+        {
+            var workspace = document.Project.Solution.Workspace;
+            var tree = await document.GetSyntaxTreeAsync(cancellationToken).ConfigureAwait(false);
+            return GetSyntacticClassifications(tree, span, workspace, cancellationToken);
+        }
+
+        public ImmutableArray<ClassifiedSpan> GetSyntacticClassifications(SyntaxTree tree, TextSpan span, Workspace workspace, CancellationToken cancellationToken)
+        {
+            var syntaxClassifiers = this.GetDefaultSyntaxClassifiers();
+
+            var extensionManager = workspace.Services.GetService<IExtensionManager>();
+            var getNodeClassifiers = extensionManager.CreateNodeExtensionGetter(syntaxClassifiers, c => c.SyntaxNodeTypes);
+            var getTokenClassifiers = extensionManager.CreateTokenExtensionGetter(syntaxClassifiers, c => c.SyntaxTokenKinds);
+
+            var result = ImmutableArray<ClassifiedSpan>.Empty;
+            var classifications = SharedPools.Default<List<ClassifiedSpan>>().Allocate();
+
+            try
+            {
+                AddSyntacticClassifications(tree, span, classifications, cancellationToken);
+                result = classifications.ToImmutableArray();
+            }
+            catch (Exception e) when (FatalError.ReportUnlessCanceled(e))
+            {
+                throw ExceptionUtilities.Unreachable;
+            }
+            finally
+            {
+                SharedPools.Default<List<ClassifiedSpan>>().ClearAndFree(classifications);
+            }
+
+            return result;
+        }
+
+        protected virtual void AddSyntacticClassifications(SyntaxTree syntaxTree, TextSpan textSpan, List<ClassifiedSpan> result, CancellationToken cancellationToken)
+        {
+        }
+
+        public override async Task<ImmutableArray<ClassifiedSpan>> GetSemanticClassificationsAsync(Document document, TextSpan span, CancellationToken cancellationToken)
+        {
+            var workspace = document.Project.Solution.Workspace;
+            var model = await document.GetSemanticModelForSpanAsync(span, cancellationToken).ConfigureAwait(false);
+            return this.GetSemanticClassifications(model, span, workspace, cancellationToken);
+        }
+
+        public ImmutableArray<ClassifiedSpan> GetSemanticClassifications(SemanticModel model, TextSpan span, Workspace workspace, CancellationToken cancellationToken)
+        {
+            var syntaxClassifiers = this.GetDefaultSyntaxClassifiers();
+
+            var extensionManager = workspace.Services.GetService<IExtensionManager>();
+            var getNodeClassifiers = extensionManager.CreateNodeExtensionGetter(syntaxClassifiers, c => c.SyntaxNodeTypes);
```

Comment:
we're using the SyntaxClassifiers to create SemanticClassifications?  Or are SemanticClassifications a superset of SyntaxClassifications here?